cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env python3
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Look through one or more folders for Cylc 7 syntax ``suite*.rc`` files
+and search for syntax which may be problematic at Cylc 8.
+
+Can be run either as a linter or "in place" (``-i``), leaving comments
+in files. If used in the "in place" mode it is recommended that you ensure
+that you have recorded the state of your workflow in a version control
+system before starting.
+"""
+from colorama import Fore
+from optparse import Values
+from pathlib import Path
+import re
+from typing import Generator
+
+from cylc.flow import LOG
+from cylc.flow.option_parsers import (
+    CylcOptionParser as COP,
+)
+from cylc.flow.terminal import cli_function
+
+STYLE_GUIDE = (
+    'https://cylc.github.io/cylc-doc/latest/html/workflow-design-guide/'
+    'style-guide.html#'
+)
+URL_STUB = "https://cylc.github.io/cylc-doc/latest/html/7-to-8/"
+SECTION1 = r'\[{}\]'
```

Comment:
Spaces between the brackets and the text are valid, so may want to include that in the regexes
```suggestion
SECTION1 = r'\[\s*{}\s*\]'
```